http-clients/java-nio-client/src/test/java/software/amazon/awssdk/http/nio/java/internal/S3JavaHttpClientIntegratedTest.java

```diff
@@ -0,0 +1,159 @@
+package software.amazon.awssdk.http.nio.java.internal;
+
+
+import static junit.framework.TestCase.assertEquals;
+import static org.apache.commons.lang3.RandomStringUtils.randomAlphabetic;
+import java.util.concurrent.CompletableFuture;
+import org.junit.Test;
+import software.amazon.awssdk.core.ResponseBytes;
+import software.amazon.awssdk.core.async.AsyncRequestBody;
+import software.amazon.awssdk.core.async.AsyncResponseTransformer;
+import software.amazon.awssdk.core.retry.RetryPolicy;
+import software.amazon.awssdk.http.Protocol;
+import software.amazon.awssdk.http.async.SdkAsyncHttpClient;
+import software.amazon.awssdk.http.nio.java.JavaHttpClientNioAsyncHttpClient;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.s3.S3AsyncClient;
+import software.amazon.awssdk.services.s3.S3Configuration;
+import software.amazon.awssdk.services.s3.model.*;
+
+public class S3JavaHttpClientIntegratedTest {
+
+    private static final String BUCKET = "zhouquas3testbucketjava";
+    private static final String KEY = "smalltestkey";
+    private static final String LARGEKEY = "largetestkey";
+    private static final String VALUE = "Hello World!";
+    private static String LARGESTRING = randomAlphabetic(25000);
+
+    private S3AsyncClient s3Async;
+
+    private void createS3AsyncClient(SdkAsyncHttpClient myClient) {
+        this.s3Async = S3AsyncClient.builder()
+                .httpClient(myClient)
+                .overrideConfiguration(b -> b.retryPolicy(RetryPolicy.none()))
+                .serviceConfiguration(S3Configuration.builder().checksumValidationEnabled(false).build())
+                .region(Region.US_WEST_2)
+                .build();
+    }
+
+    private void createJavaHttpClientHTTP1() {
+        SdkAsyncHttpClient javaHttpCient = JavaHttpClientNioAsyncHttpClient.builder().protocol(Protocol.HTTP1_1).build();
+        createS3AsyncClient(javaHttpCient);
+    }
+
+    private void createJavaHttpClientHTTP2() {
+        SdkAsyncHttpClient javaHttpCient = JavaHttpClientNioAsyncHttpClient.builder().protocol(Protocol.HTTP2).build();
+        createS3AsyncClient(javaHttpCient);
+    }
+
+    @Test
+    public void s3creatBucketWithJavaHttpClientHTTP1Test() {
+        createJavaHttpClientHTTP1();
+        createBucketTest(s3Async);
+    }
+
+    @Test
+    public void s3creatBucketWithJavaHttpClientHTTP2Test() {
+        createJavaHttpClientHTTP2();
+        createBucketTest(s3Async);
+    }
+
+    @Test
+    public void s3PutSmallObjectWithJavaHttpClientHTTP1Test() {
+//        System.setProperty("jdk.internal.httpclient.debug", "true");
+//        System.setProperty("jdk.internal.httpclient.websocket.debug", "true");
+        createJavaHttpClientHTTP1();
+        putObjectTest(s3Async, KEY, VALUE);
+    }
+
+    @Test
+    public void s3PutSmallObjectWithJavaHttpClientHTTP2Test() {
+        createJavaHttpClientHTTP2();
+        putObjectTest(s3Async, KEY, VALUE);
```

Comment:
To verify, we should have a second S3 client (that doesn't use the Java Nio client) that downloads the object and checks it. Let me know if you need help with this.

Reply:
Can I add a new test file in S3 module? Since if I have to use another client I will have to add new dependency on Netty.

Reply:
You don't need to use the netty client. You can use a the sync version of the S3 client; we just need to to check that the file in S3 is correct; so something like

```java
S3Client s3 = S3Client.create();

// call getObject()
// Check the contents of the downloaded object to make sure it's correct by comparing the MD5 checksums.
```